java/test/org/openqa/selenium/environment/webserver/CookieHandler.java

```diff
@@ -191,4 +191,16 @@ private Cookie parse(String cookieString) {
 
     return builder.build();
   }
+  private String escapeCookieValue(String value) {
+    if (value == null) {
+      return "";
+    }
+
+    return value.replace("\\", "\\\\")
+      .replace("\"", "\\\"")
+      .replace(";", "\\;")
+      .replace(",", "\\,")
+      .replace("\r", "")
+      .replace("\n", "");
```

Comment:
These characters are escaped and are commonly used in cookie name and value.

I haven't added escaping for XSS Cookie Handling because it is rare to have those when using a HTTP Handler. Can add if required.

Reply:
@harsha509 suggested to add XSS Cookie Handling for an extra security layer.